fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1186,8 +1188,33 @@ private ScoredPlan planOr(@Nonnull PlanContext planContext, @Nonnull OrComponent
             if (subplan.planOrderingKey == null) {
                 allHaveOrderingKey = false;
             }
+            RecordQueryPlan filteredBasePlan;
+            if (subplan.plan instanceof RecordQueryFilterPlan) {
+                filteredBasePlan = ((RecordQueryFilterPlan)subplan.plan).getInner();
+            } else {
+                filteredBasePlan = null;
+            }
+            if (subplans.isEmpty()) {
+                commonFilteredBasePlan = filteredBasePlan;
+                allHaveSameBasePlan = filteredBasePlan != null;
+            } else if (allHaveSameBasePlan && !Objects.equals(filteredBasePlan, commonFilteredBasePlan)) {
+                allHaveSameBasePlan = false;
+            }
             subplans.add(subplan);
         }
+        // If the child plans only differ in their filters, then there is no point in repeating the base
+        // scan only to evaluate each of the filters. Just evaluate the scan with an OR filter.
+        // Note that this also improves the _second-best_ plan for planFilterWithInJoin, but an IN filter wins
+        // out there over the equivalent OR(EQUALS) filters.
```

Comment:
Specifically, for `FDBInQueryTest.testInQueryIndexSortedDifferently`, the old alternative plan was
```
Index(MySimpleRecord$str_value_indexed <,>) | num_value_3_indexed EQUALS 1 ∪[Field { 'str_value_indexed' None}, Field { 'rec_no' None}] Index(MySimpleRecord$str_value_indexed <,>) | num_value_3_indexed EQUALS 4 ∪[Field { 'str_value_indexed' None}, Field { 'rec_no' None}] Index(MySimpleRecord$str_value_indexed <,>) | num_value_3_indexed EQUALS 2
```
and the new one is
```
Index(MySimpleRecord$str_value_indexed <,>) | Or([num_value_3_indexed EQUALS 1, num_value_3_indexed EQUALS 4, num_value_3_indexed EQUALS 2])
```
but the winner still is
```
Index(MySimpleRecord$str_value_indexed <,>) | num_value_3_indexed IN [1, 4, 2]
```
Hence no change in that test.